src/c++/perf_analyzer/CMakeLists.txt

```diff
@@ -136,12 +136,15 @@ add_executable(
   perf_analyzer_unit_tests
   ${PERF_ANALYZER_UNIT_TESTS_SRCS}
   ${PERF_ANALYZER_UNIT_TESTS_HDRS}
+  mock_inference_profiler.h
```

Comment:
I would prefer if all testing code was placed in the implementation file, as the doctest methodology recommends:

[https://github.com/doctest/doctest](https://github.com/doctest/doctest):

> This allows the framework to be used in more ways than any other - tests can be written directly in the production code!
> 
> *Tests can be a form of documentation and should be able to reside near the production code which they test.*
> 
> - This makes the barrier for writing tests **much lower** - you don't have to: **1)** make a separate source file **2)** include a bunch of stuff in it **3)** add it to the build system and **4)** add it to source control - You can just write the tests for a class or a piece of functionality at the bottom of its source file - or even header file!
> - Tests in the production code can be thought of as documentation/up-to-date comments - showcasing the APIs
> - Testing internals that are not exposed through the public API and headers is no longer a mind-bending exercise
> - [**Test-driven development**](https://en.wikipedia.org/wiki/Test-driven_development) in C++ has never been easier!
> 
> The framework can be used just like any other without mixing production code and tests - check out the [**features**](doc/markdown/features.md).

Reply:
These files are already too long to parse reasonably. If these files were shorter, I would agree with you.

Reply:
Tests in the same file don't really make sense if the file is massive. We can consider putting testing in the same file once we have refactored this into many smaller files.